packages/wrangler/src/index.tsx

```diff
@@ -1922,8 +1922,35 @@ function createCLIParser(argv: string[]) {
 							});
 							logger.log("Upload complete.");
 						}
+					)
+					.command(
+						"delete <objectPath>",
+						"Delete an object in an R2 bucket",
+						(Objectyargs) => {
+							return Objectyargs.positional("objectPath", {
+								describe:
+									"The destination object path in the form of {bucket}/{key}",
+								type: "string",
+							});
+						},
+						async (args) => {
+							const { objectPath } = args;
+							await printWranglerBanner();
+
+							const config = readConfig(args.config as ConfigPath, args);
+							const accountId = await requireAuth(config);
+							const { bucket, key } = bucketAndKeyFromObjectPath(objectPath);
+							logger.log(`Deleting object "${key}" from bucket "${bucket}".`);
+							try {
+								await deleteR2Object(accountId, bucket, key);
+								logger.log("Delete complete.");
+							} catch (error) {
+								logger.error(`Delete failed. ${error}`);
+							}
```

Comment:
I think it would be better to translate the API error into a user-friendly error, and just throw that instead of using this try/catch.

Reply:
I will consider improving the handling for the pattern of these type of fetches for KV and R2